utils.py

```diff
@@ -134,19 +136,28 @@ def shuffle_dataset(self, random_seed=None):
         self.image_filenames = self.image_filenames[idx]
         self.label_filenames = self.label_filenames[idx]
 
+    def next_raw_data(self):
+        index = self.current_index % self.dataset_size
+        image_filename = self.image_filenames[index]
+        label_filename = self.label_filenames[index]
+        self.current_index = index + 1
+        image = read_image(image_filename=image_filename)
+        label = read_label(label_filename=label_filename)
+        label = np.expand_dims(label, axis=2)
+
+        return image, label
+
     def next_minibatch(self):
 
-        image_filenames_minibatch = self.image_filenames[self.current_index: self.current_index + self.minibatch_size]
-        label_filenames_minibatch = self.label_filenames[self.current_index: self.current_index + self.minibatch_size]
-        self.current_index += self.minibatch_size
-        if self.current_index > self.dataset_size:
-            self.current_index = 0
+        minibatch_indices = np.arange(self.current_index, self.current_index + self.minibatch_size) % self.dataset_size
```

Comment:
This implementation of next_minibatch() will go over one epoch when you intended to just do one epoch, while my old implementation will just do exact one epoch if you intended to do just one epoch.